src/OrchardCore.Modules/OrchardCore.Rules/Services/RoleConditionEvaluator.cs

```diff
@@ -31,6 +31,17 @@ public override ValueTask<bool> EvaluateAsync(RoleCondition condition)
             // IsInRole() & HasClaim() are case sensitive.
             var operatorComparer = _operatorResolver.GetOperatorComparer(condition.Operation);
 
+            // Claim all if the operator is negative
+            if(condition.Operation == new StringNotEqualsOperator() ||
```

Comment:
I agree with the choice to use `.All` 

What I would suggest is to add an interface to the negative operators, `INegateOperator` use something like
`if (condition.Operation is INegateOperator)...`